domain/snyk/scanner/scanner.go

```
@@ -294,13 +296,31 @@ func (sc *DelegatingConcurrentScanner) Scan(
 					TimestampFinished: time.Now().UTC(),
 					Path:              folderPath,
 				}
-
+				deltaScanEnabled, deltaScanner := isDeltaScanEnabled(s)
 				// in case of delta scans, we add additional fields
-				if enabled, deltaScanner := isDeltaScanEnabled(s); enabled {
+				if deltaScanEnabled {
 					data.IsDeltaScan = deltaScanner.DeltaScanningEnabled()
 				}
 
-				processResults(data)
+				processResults(data, true, true)
+				go func() {
+					defer referenceBranchScanWaitGroup.Done()
+					// TODO: implement proper context handling
+					sc.scanStateAggregator.SetScanInProgress(folderPath, scanner.Product(), true)
+					err := sc.scanBaseBranch(context.Background(), s, folderPath, gitCheckoutHandler)
+					sc.scanStateAggregator.SetScanDone(folderPath, scanner.Product(), true, err)
+					// TODO: is this a good idea?
+					data = snyk.ScanData{
+						Product: s.Product(),
+						Path:    gitCheckoutHandler.BaseFolderPath(),
+					}
+					processResults(data, false, false)
+					// TODO: where should we report errors ?
+					if err != nil {
+						logger.Error().Err(err).Msgf("couldn't scan base branch for folder %s for product %s", folderPath, s.Product())
```

Probably we should send `snyk.scan` for reference scans, so we can report these errors to the user and take action, e.g. disable delta scanning.

---

We could also send an error notification with the notifier to the user.

---

Currently this the case. Since processResults will call publishDiagnostics and could potentially send a snyk error